spacy_llm/tasks/util/span.py

```diff
@@ -36,6 +36,27 @@ def __init__(
         self._case_sensitive_matching = case_sensitive_matching
         self._single_match = single_match
 
+        self._check_label_consistency()
+
+    def _check_label_consistency(self) -> None:
+        """Checks consistency of labels between examples and defined labels."""
+        if not self._examples:
+            return
+        example_labels = list(
+            {
+                self._normalizer(key)
+                for example in self._examples
+                for key in example.entities
+            }
+        )
+        spec_labels = list(self._label_dict.keys())
+        example_labels.sort()
+        spec_labels.sort()
+        if not set(example_labels) <= set(spec_labels):
+            raise ValueError(
+                f"Specified labels and labels in examples diverge. Received {spec_labels} and {example_labels}, correspondingly. Please ensure your label specification and example labels are consistent."
```

Comment:
I would make this warning message more specific and say 
1) which label set was defined as part of the task definition 
2) which labels were found in the examples that were NOT part of the task definition.

We can compile list 2) by keeping track of the labels not in `spec_labels`.

This will ultimately help users understand more quickly what is going on and which parts of their configuration they need to adjust.

I think this approach will also allow us to output the actual original labels instead of the normalized ones. We'd store the original in the deviating list if it's normalized form is not in `spec_labels`.

Reply:
Like in https://github.com/explosion/spacy-llm/pull/183/commits/39606aaefbea9950369f4d2a6ea2a65a21c1021c?